### PR TITLE
sp-std -> core

### PR DESCRIPTION
**Pull request by gilescope**
First in a series of PRs that reduces our use of sp-std with a view to deprecating it.

This is just looking at /substrate and moving some of the references from `sp-std` to `core`.
These particular changes should be uncontroversial.

Where macros are used `::core` should be used to remove any ambiguity.

part of https://github.com/paritytech/polkadot-sdk/issues/2101

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5115687

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5115686

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5115688

File: prdoc/pr_3199.prdoc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: sp-std to core
+
+doc:
+  - audience: Node Dev
```
Comment (kianenigma) [inline prdoc/pr_3199.prdoc:7]: Runtime devs also care about this afaik?

**Review by kianenigma — APPROVED** [review]
I am glad we are doing this step by step. 

As noted in https://github.com/paritytech/polkadot-sdk/pull/3199/files, I am not super convinced that the whole deprecation is a winning tradeoff, but I am too late to that conversation 🎩

File: prdoc/pr_3199.prdoc (hunk 1 above)
Comment (bkchr) [reply prdoc/pr_3199.prdoc:7]: This doesn't require any prdoc at all. This is a purely internal change that doesn't change anything to the outside.

Comment (kianenigma) [reply prdoc/pr_3199.prdoc:7]: Yes, good point. Only when `sp_std` itself is removed or marked deprecated we would need a prdoc and possibly some doc updates.

Comment (bkchr) [reply prdoc/pr_3199.prdoc:7]: Exactly.

**Comment by gilescope** [comment]
There is merit in just doing all the core references (there are a lot of `extern crate alloc` references that would have to be put almost *everywhere* if we do the `alloc`).

File: prdoc/pr_3199.prdoc (hunk 1 above)
Comment (gilescope) [reply prdoc/pr_3199.prdoc]: Have removed prdoc.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5125201

**Comment by gilescope** [comment]
ok that set of changes looks good to me.

**Review by davxy — APPROVED** [review]

**Review by koute — APPROVED** [review]

**Pull request merged**